polaris-service/src/main/java/org/apache/polaris/service/task/ManifestFileCleanupTaskHandler.java

```diff
@@ -68,58 +70,107 @@ public boolean canHandleTask(TaskEntity task) {
   @Override
   public boolean handleTask(TaskEntity task) {
     ManifestCleanupTask cleanupTask = task.readData(ManifestCleanupTask.class);
-    ManifestFile manifestFile = decodeManifestData(cleanupTask.getManifestFileData());
     TableIdentifier tableId = cleanupTask.getTableId();
     try (FileIO authorizedFileIO = fileIOSupplier.apply(task)) {
-
```

Comment:
Combine manifest and content file cleanup in the same task, but ensure that each task processes only one type of data (either manifest or content files) at a time.